### PR TITLE
Propagate Python workers' error logs to Error Reporting

### PR DESCRIPTION
**Pull request by michaelkedar**
Part of #1053

### CODE REVIEW

**Review by another-rex — COMMENTED** [review]
Maybe it'll be worth it to pull the `_setup_gcp_logging` out into it's own library? Avoids duplicate code, and appengine / API / other google cloud pieces can use it as well.

**Comment by michaelkedar** [comment]
> Maybe it'll be worth it to pull the `_setup_gcp_logging` out into it's own library? Avoids duplicate code, and appengine / API / other google cloud pieces can use it as well.

Good point. I've created `osv.logging_helper` and put it there.

**Review by another-rex — APPROVED** [review]
LGTM

File: osv/logging_helper.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                  'functionName': record.funcName,
+              }
+          },
+      })
+
+    return record
+
+  logging.setLogRecordFactory(record_factory)
+  logging.getLogger().setLevel(logging.INFO)
+  logging.getLogger('google.api_core.bidi').setLevel(logging.ERROR)
```
Comment (oliverchang) [inline osv/logging_helper.py:58]: As a bit of documentation improvement while we're here, can you please add comments here to explain what these are doing? (i.e. suppressing noisy logs in our dependencies?)

File: osv/logging_helper.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Google LLC
```
Comment (oliverchang) [inline osv/logging_helper.py:1]: nit: can we just name this module "logging" ? "helper" is generally not a very descriptive word for naming.

**Review by oliverchang — APPROVED** [review]
thanks!! just some nits.

File: osv/logging_helper.py (hunk 2 above)
Comment (michaelkedar) [reply osv/logging_helper.py:1]: I could do that, but I'm a bit wary of some ambiguity - osv's `__init__.py` imports everything from a few files, including the python `logging` module. So you end up with something like
```python
>>> import osv
>>> osv.logging
"<module 'logging' from '/usr/lib/python3.11/logging/__init__.py'>"
>>> import osv.logging
>>> osv.logging
"<module 'osv.logging' from '/path/to/osv.dev/osv/logging.py'>"
```

(I actually forgot to import `osv.logging_helper` anyway - will fix that)

Comment (oliverchang) [reply osv/logging_helper.py:1]: Ah, very good point here. 

In that case, can we name this something like "logs" instead?

File: osv/logging_helper.py (hunk 1 above)
Comment (michaelkedar) [reply osv/logging_helper.py]: Added that comment.

File: osv/logging_helper.py (hunk 2 above)
Comment (michaelkedar) [reply osv/logging_helper.py]: Changed to `osv.logs`

**Review by oliverchang — APPROVED** [review]

**Pull request merged**